src/layers/core/scatterplot-layer/scatterplot-layer.js

```diff
@@ -65,9 +72,47 @@ export default class ScatterplotLayer extends Layer {
       instanceRadius: {size: 1, accessor: 'getRadius', defaultValue: 1, update: this.calculateInstanceRadius},
       instanceColors: {size: 4, type: GL.UNSIGNED_BYTE, accessor: 'getColor', update: this.calculateInstanceColors}
     });
+
+    if (this.props.fp64) {
+      this.state.attributeManager.addInstanced({
+        instancePositions64xyLow: {size: 2, accessor: 'getPosition', update: this.calculateInstancePositions64xyLow}
+      });
+    }
     /* eslint-enable max-len */
   }
 
+  updateModel({props, oldProps, changeFlags}) {
+    if (props.fp64 !== oldProps.fp64) {
+      const {gl} = this.context;
+      this.setState({model: this._getModel(gl)});
+    }
+  }
+
+  updateAttribute({props, oldProps, changeFlags}) {
+    if (props.fp64 !== oldProps.fp64) {
+      const {attributeManager} = this.state;
+
+      // all attributes needs to be set again after the model changes
```

Comment:
@ibgreen This is how attribute manager works now, and I agree that this sounds like a bug to me.
Right now, luma.gl's render function get buffers to bind to attributes from Model.buffers, whose content is populated in model.setAttributes() function. Model.setAttributes() is in turn called by Layer.updateAttributes() with attributes dictionary generated by attributeManager.getChangedAttributes().

So only attributes with changed data will be set to Model. In scatterplot's case, changing between 32-bit and 64-bit model does not change the underlying attribute data, so the attribute manager failed to populate those attributes to new model. We have to do this manually instead.

Be noted that this is not the case for polygon layer as the polygon tessellator regenerate the underlying data when 32-bit and 64-bit change happens so all attributes got automatically populated.